citre-lang-c.el

```diff
@@ -86,17 +86,22 @@
   "Get non-header symbol at point."
   (when-let ((symbol (citre-get-symbol-default)))
     (let* ((bounds (citre-get-property 'bounds symbol))
+           callable
            syntax)
       (save-excursion
         (goto-char (car bounds))
         (when (looking-back (rx (or "->" ".")) (- (point) 2))
           (setq syntax 'member)))
-      (unless syntax
-        (save-excursion
-          (goto-char (cdr bounds))
-          (when (looking-at (rx (* " ") "("))
-            (setq syntax 'function))))
-      (citre-put-property symbol 'syntax syntax))))
+      (save-excursion
+        (goto-char (cdr bounds))
+        (when (looking-at (rx (* " ") "("))
+          (cond
+           ((eq syntax 'member)
+            (setq callable t))
+           ((not syntax)
+            (setq syntax 'function)))))
+      (citre-put-property symbol 'syntax syntax
+                          'callable-member callable))))
```

Comment:
How about having 3 kinds of syntax: `member`, `callable-member` and `function`? So we can have simpler logic.